splinter/driver/webdriver/setup.py

```diff
@@ -0,0 +1,95 @@
+from selenium.webdriver import Remote
+
+"""The following functions prepare Webdriver for use.
+
+Config, Options, Service, and other arguments are configured and sent to
+the driver.
+
+The driver is returned.
+"""
+
+
+def _setup_chrome(driver_class, config=None, options=None, service=None, **kwargs):
```

Comment:
Do these need to default to `None`? we're always providing it, no?

Reply:
I tend to prefer keyword arguments with None as a default to positional arguments. It's just personal preference, it doesn't make a difference for a private function like this.

Because this moves almost all the logic out of the Webdriver sub-classes, there's probably a more evolved version of these functions that have a closer relationship to the Config and/or Browser objects. If there's no reason to sub-class splinter's webdriver class, we'll need to build an interface for handling Config to driver communication if we want to expose any of this to the public.

Reply:
Oh, the reason this caught my attention is that the code in the function clearly assumes that `config` is never `None` and that `options` should probably never be `None`, but the function signature indicates that they are usually `None`. It's not just a matter of API, but rather making it clear to the reader what they can expect from this function without having to dig into details. When I see a default value, I expect the code to work with it.

Reply:
That's not wrong, but a None value will cause an error where it fails whereas a missing positional argument will cause a TypeError on the function call. I find the latter less informative because it only tells you that something is missing. The former lets you know that something is missing and something downstream is exploding as a result. I treat it on a situation-by-situation basis. If I'm writing something as an interface, I'd err on the side of the TypeError on the function call. I don't mind changing it.

Reply:
A good static analysis tool should be able to catch the missing argument, no need to wait for a runtime exception. I definitely don't think we need to block on this though, it's a private API and we can change it at anytime. Feel free to ship!